### PR TITLE
DOC: Simplify and update git setup page

### PR DESCRIPTION
**Pull request by bjnath**
Changed title from "Getting started with Git development"
to more specific "Setting up git for NumPy development"

Updated the outdated GitHub screenshot for forking button, added
more screenshots.

Simplified the narrative.

Fixed bad github links (always pointed to NumPy's github).

Clarified SSH setup, role, and effect on git remote -v.

Clarified meaning of "upstream" and "origin".

Replaced read of .git/config (platform-specific) with call to git config.

### CODE REVIEW

**Review by seberg — APPROVED** [review]
LGMT, I wonder if we should mention how to create and push a feature branch and keep the master branch up to date? In any case, I think its a big improvement, and I am happy to merge it pretty much as is.

File: doc/source/dev/gitwash/development_setup.rst [hunk 1]
```diff
@@ -1,145 +1,172 @@
...
+   default: ::
 
-   Now you want to connect to the upstream `NumPy github`_ repository, so
-   you can merge in changes from trunk.
+    git config branch.master.remote upstream
+    git config branch.master.merge refs/heads/master
 
-.. _linking-to-upstream:
+#. Optionally, to get easy access to NumPy pull requests (useful mostly to
+   reviewers): ::
```
[the remark this answers is not in the document]
Comment (bjnath) [reply doc/source/dev/gitwash/development_setup.rst:95]: No need to keep it, then -- removed.

File: doc/source/dev/gitwash/development_setup.rst [hunk 2]
```diff
@@ -1,145 +1,172 @@
...
+    remote: Counting objects: 100% (12/12), done.
+    remote: Compressing objects: 100% (12/12), done.
+    remote: Total 175837 (delta 0), reused 0 (delta 0), pack-reused 175825
+    Receiving objects: 100% (175837/175837), 78.16 MiB | 9.87 MiB/s, done.
+    Resolving deltas: 100% (139317/139317), done.
 
-In detail
-=========
+   A directory ``numpy`` is created on your machine. (If you already have
+   a numpy directory, GitHub will choose a different name like ``numpy-1``.)
```
[the remark this answers is not in the document]
Comment (bjnath) [reply doc/source/dev/gitwash/development_setup.rst:76]: In this case I think it really is GitHub and not git. If you try to use git to do what GitHub is doing and clone a repo with a name you already have, git fails with `destination path xxx already exists and is not an empty directory.`

**Comment by bjnath** [comment]
>  I wonder if we should mention how to create and push a feature branch and keep the master branch up to date?

Yes -- this is development_workflow.rst, which I'll also take a look at. It references this page as a prerequisite.

> In any case, I think its a big improvement, and I am happy to merge it pretty much as is.

Thank you! Pushed a new version that removes the advice on pull branches. Left the GitHub -> git change (https://github.com/numpy/numpy/pull/16671#discussion_r445788167) alone -- see comment.

**Comment by bjnath** [comment]
The site has overlapping content on this subject (the first three bullets of https://numpy.org/devdocs/dev/gitwash/index.html and the start of https://numpy.org/devdocs/dev/index.html#development-process-summary). Noticed it just now. I'll submit another PR to clean that up.

File: doc/source/dev/gitwash/development_setup.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (seberg) [reply doc/source/dev/gitwash/development_setup.rst:86]: Oh, interesting that the remote part has something to do with it

**Pull request merged**